xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/AbstractXHTMLLinkTypeRenderer.java

```diff
@@ -136,6 +142,31 @@ protected String computeLabel(ResourceReference reference)
         return label;
     }
 
+    /**
+     * Default implementation for computing a link title when no title has been specified. Can be overwritten by
+     * implementations to provide a different algorithm.
+     * @param reference the reference of the link for which to compute the label
+     * @return the computed title
+     * @since 15.2RC1
+     */
+    protected String computeCreateTitle(ResourceReference reference)
+    {
+        // Look for a component implementing URITitleGenerator with a role hint matching the link scheme.
+        // If not found then use the full reference as the label.
+        // If there's no scheme separator then use the full reference as the title. Note that this can happen
+        // when we're not in wiki mode (since all links are considered URIs when not in wiki mode).
+        String title;
+        try {
+            URITitleGenerator uriTitleGenerator =
+                this.componentManager.getInstance(URITitleGenerator.class, reference.getType().getScheme());
+            title = uriTitleGenerator.generateCreateTitle(reference);
+        } catch (ComponentLookupException e) {
+            e.printStackTrace();
```

Comment:
We never handle exceptions with a `printStackTrace`: you should instead log an understandable error message, using something like: 
```
this.logger.error("Error while loading component for generating URI title: [{}]", ExceptionsUtils.getRootCauseMessage(e));
this.logger.debug("Full stack trace: ", e);
```
The logger debug is not necessarily needed.